dao/src/main/scala/za/co/absa/enceladus/dao/RestClient.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.dao
+
+import org.slf4j.LoggerFactory
+import org.springframework.http._
+import org.springframework.web.client.RestTemplate
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+
+protected class RestClient(private[dao] val authClient: AuthClient,
+                           private[dao] val restTemplate: RestTemplate) {
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  private[dao] var authHeaders: HttpHeaders = new HttpHeaders()
```

Comment:
You're right about the access modifier, but I wouldn't use a new public function to give it an accessor (it would also need a mutator) because it would only be used in the tests. I think it would be cleaner with a bit of reflection instead.